@garymathews , Following the same steps as my above comment with the recent fix, now I get a crash:
```
[ERROR] :  TiApplication: (main) [6561,6561] Sending event: exception on thread: main msg:java.lang.RuntimeException: Unable to start activity ComponentInfo{com.app.timob15253/org.appcelerator.titanium.TiActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.Activity ti.modules.titanium.ui.TabGroupProxy.getActivity()' on a null object reference; Titanium 6.1.0,2017/03/02 15:59,undefined
[ERROR] :  TiApplication: java.lang.RuntimeException: Unable to start activity ComponentInfo{com.app.timob15253/org.appcelerator.titanium.TiActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.Activity ti.modules.titanium.ui.TabGroupProxy.getActivity()' on a null object reference
[ERROR] :  TiApplication: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2416)
[ERROR] :  TiApplication: 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2476)
[ERROR] :  TiApplication: 	at android.app.ActivityThread.-wrap11(ActivityThread.java)
[ERROR] :  TiApplication: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
[ERROR] :  TiApplication: 	at android.os.Handler.dispatchMessage(Handler.java:102)
[ERROR] :  TiApplication: 	at android.os.Looper.loop(Looper.java:148)
[ERROR] :  TiApplication: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
[ERROR] :  TiApplication: 	at java.lang.reflect.Method.invoke(Native Method)
[ERROR] :  TiApplication: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
[ERROR] :  TiApplication: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
[ERROR] :  TiApplication: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.Activity ti.modules.titanium.ui.TabGroupProxy.getActivity()' on a null object reference
[ERROR] :  TiApplication: 	at ti.modules.titanium.ui.widget.tabgroup.TiUIAbstractTab.getContentView(TiUIAbstractTab.java:51)
[ERROR] :  TiApplication: 	at ti.modules.titanium.ui.widget.tabgroup.TiUIActionBarTab$TabFragment.onCreateView(TiUIActionBarTab.java:46)
[ERROR] :  TiApplication: 	at android.support.v4.app.Fragment.performCreateView(Fragment.java:1962)
[ERROR] :  TiApplication: 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1067)
[ERROR] :  TiApplication: 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1248)
[ERROR] :  TiApplication: 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1230)
[ERROR] :  TiApplication: 	at android.support.v4.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2042)
[ERROR] :  TiApplication: 	at android.support.v4.app.FragmentController.dispatchActivityCreated(FragmentController.java:165)
[ERROR] :  TiApplication: 	at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:543)
[ERROR] :  TiApplication: 	at org.appcelerator.titanium.TiBaseActivity.onStart(TiBaseActivity.java:1346)
[ERROR] :  TiApplication: 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1237)
[ERROR] :  TiApplication: 	at android.app.Activity.performStart(Activity.java:6268)
[ERROR] :  TiApplication: 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2379)
[ERROR] :  TiApplication: 	... 9 more
[INFO] :   Process: Sending signal. PID: 5639 SIG: 9
```

--- a/android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIAbstractTab.java
+++ b/android/modules/ui/src/java/ti/modules/titanium/ui/widget/tabgroup/TiUIAbstractTab.java
@@ -38,7 +38,7 @@ public abstract class TiUIAbstractTab extends TiUIView {
 	 */
 	public View getContentView() {
 		TiWindowProxy windowProxy = getWindowProxy();
-		if (windowProxy == null) {
+		if (windowProxy == null || proxy == null) {
 			// If no window is provided use an empty view.
 			View emptyContent = new View(TiApplication.getInstance().getApplicationContext());
 			emptyContent.setBackgroundColor(Color.BLACK);

--- a/android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java
+++ b/android/titanium/src/java/org/appcelerator/titanium/TiBaseActivity.java
@@ -863,13 +863,17 @@ public abstract class TiBaseActivity extends AppCompatActivity
 			if (windowStack.size() <= 1) {
 				if (topWindow != null) {
 					if (TiConvert.toBoolean(topWindow.getProperty(TiC.PROPERTY_EXIT_ON_CLOSE), true)) {
-						Log.d(TAG, "onBackPressed: exit", Log.DEBUG_MODE);
-						TiApplication.terminateActivityStack();
+						Log.d(TAG, "onBackPressed: exit");
+						if (Build.VERSION.SDK_INT >= 16) {
+							finishAffinity();
+						} else {
+							TiApplication.terminateActivityStack();
+						}
 					} else {
-						Log.d(TAG, "onBackPressed: suspend to background", Log.DEBUG_MODE);
+						Log.d(TAG, "onBackPressed: suspend to background");
 						this.moveTaskToBack(true);
-						return;
 					}
+					return;
 				}
  			}
 

